pydantic/config.py

```diff
@@ -34,189 +35,167 @@ class Extra(str, Enum):
-            alias = cls.alias_generator(name)
-            if not isinstance(alias, str):
-                raise TypeError(f'Config.alias_generator must return str, not {alias.__class__}')
-            field_info.update(alias=alias, alias_priority=1)
-        return field_info
-
-    @classmethod
-    def prepare_field(cls, field: Any) -> None:
-        """
-        Optional hook to check or modify fields during model creation.
-        """
-        pass
-
-
-def get_config(config: Union[ConfigDict, Type[object], None]) -> Type[BaseConfig]:
-    if config is None:
-        return BaseConfig
+    ser_json_timedelta: Literal['iso8601', 'float']
+    ser_json_bytes: Literal['utf8', 'base64']
 
-    else:
-        config_dict = (
-            config
-            if isinstance(config, dict)
-            else {k: getattr(config, k) for k in dir(config) if not k.startswith('__')}
-        )
 
-        class Config(BaseConfig):
-            ...
+config_keys = set(_ConfigDict.__annotations__.keys())
 
-        for k, v in config_dict.items():
-            setattr(Config, k, v)
-        return Config
 
+if TYPE_CHECKING:
 
-def inherit_config(self_config: 'ConfigType', parent_config: 'ConfigType', **namespace: Any) -> 'ConfigType':
-    # TODO remove
-    if not self_config:
-        base_classes: Tuple['ConfigType', ...] = (parent_config,)
-    elif self_config == parent_config:
-        base_classes = (self_config,)
-    else:
-        base_classes = self_config, parent_config
+    class ConfigDict(_ConfigDict):
+        ...
+
+else:
+
+    class ConfigDict(dict):
+        def __missing__(self, key):
+            return _default_config[key]
+
+
+_default_config = ConfigDict(
+    title=None,
+    str_to_lower=False,
+    str_to_upper=False,
+    str_strip_whitespace=False,
+    str_min_length=0,
+    str_max_length=None,
+    extra=Extra.ignore,
+    frozen=False,
+    populate_by_name=False,
+    use_enum_values=False,
+    validate_assignment=False,
+    arbitrary_types_allowed=False,
+    undefined_types_warning=True,
+    from_attributes=False,
+    alias_generator=None,
+    keep_untouched=(),
+    json_loads=json.loads,
+    json_dumps=json.dumps,
+    json_encoders={},
+    allow_inf_nan=True,
+    strict=False,
+    copy_on_model_validation='shallow',
+    post_init_call='before_validation',
+    ser_json_timedelta='iso8601',
+    ser_json_bytes='utf8',
+)
+
+
+class ConfigMetaclass(type):
+    def __getattr__(self, item: str) -> Any:
+        return _default_config.get(item)
+
+
+class BaseConfig(metaclass=ConfigMetaclass):
+    """
+    This class is only retained for backwards compatibility.
 
-    namespace['json_encoders'] = {
-        **getattr(parent_config, 'json_encoders', {}),
-        **getattr(self_config, 'json_encoders', {}),
-        **namespace.get('json_encoders', {}),
-    }
+    The preferred approach going forward is to assign a ConfigDict to the `model_config` attribute of the Model class.
```

Comment:
@samuelcolvin Are you good with this docstring? Maybe there is a better way to phrase it. (Also I'm not actually sure if what it says is true, but it seems to match what I saw in some other tests that I assumed was the preferred approach..)

Reply:
LGTM.